File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n=nums.length;
        Map<Integer,Integer> map=new HashMap<>();
        int[] r=new int[2];
        for(int i=0;i<n;i++){
            if(map.containsKey(target-nums[i])){
                r[1]=i;
                r[0]=map.get(target-nums[i]);
                return r;
            }
            map.put(nums[i],i);
        }
        return r;
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numsMap = {}
        
        for index in range(len(nums)):
            if nums[index] not in numsMap:
                numsMap[nums[index]] = index
        print(numsMap)
        for i in range(len(nums)):
            diff = target - nums[i]
            if diff in numsMap and numsMap[diff] != i:
                return[i, numsMap[diff]]
            
        return [-1,-1]
                
                
        